react/components/ProductQuantitySelector/index.js

```diff
@@ -0,0 +1,81 @@
+import React, { useCallback } from 'react'
+import PropTypes from 'prop-types'
+import { NumericStepper } from 'vtex.styleguide'
+import { FormattedMessage } from 'react-intl'
+import { ProductContext } from 'vtex.product-context'
+import { path, isEmpty } from 'ramda'
+
+const ProductQuantitySelector = ({ warningQuantityThreshold, ...props }) => {
+  const valuesFromContext = React.useContext(ProductContext)
+
+  const numericStepperProps = () => {
+    if (!valuesFromContext || isEmpty(valuesFromContext)) {
+      const { onChange, availableQuantity, selectedQuantity } = props
+
+      return {
+        availableQuantity,
+        selectedQuantity,
+        onChange: useCallback(e => onChange(e.value), []),
```

Comment:
`useCallback` inside a function? isn't the lint complaining on this?

Reply:
I think you can create the `onChange` function outside this, and use a ref to assign which function should be called

Reply:
good catch, I'll try to understand what was the intention of the author here

Reply:
removed the useCallback hook altogether, for now

Reply:
Probably it was a POC to improve performance. We were having issues with unnecessary rerenders. You can remove it anyway